### ISSUE TITLE
colnamesInt() has unfriendly behavior w.r.t. missing (NA) columns

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#6070]
It's a bit strange that `NA` returns `0L` for character input, but `NA_integer_` for numeric input.

In fact I find the current behavior odd as well:

```r
data.table:::colnamesInt(iris, NA_integer_)
# Error in data.table:::colnamesInt(iris, NA_integer_) : 
#   argument specifying columns received non-existing column(s): cols[1]=-2147483648
```

I think it's outside the scope of this PR though, I'll file a follow-up issue.

_Originally posted by @MichaelChirico in https://github.com/Rdatatable/data.table/pull/6068#discussion_r1560148562_
            
First, we should check that it's actually possible to provide NA to this column since it's a private function (maybe we fully control all inputs & know for sure it can't receive missing inputs).

If user might provide `NA`, we should handle this better.

If `NA` input is impossible, we should drop tests with missing input.

**Comment by Nj221102** [comment]
Yeah, and starting from issue #6068, its behavior with numeric input has also been updated to produce an error. Currently, if you provide an integer vector with NA entries as input to `cols`, it will result in an error.

**Comment by Nj221102** [comment]
I think making Na_integer give 0L as output will be consistent with what Na as character input do , so i think that will be good, wdyt? @MichaelChirico

**Comment by MichaelChirico** [comment]
> First, we should check that it's actually possible to provide NA to this column since it's a private function (maybe we fully control all inputs & know for sure it can't receive missing inputs).

Please start from here -- can you write an example using only the public API that generates the above "bad" error message?

**Comment by Nj221102** [comment]
> > First, we should check that it's actually possible to provide NA to this column since it's a private function (maybe we fully control all inputs & know for sure it can't receive missing inputs).
> 
> Please start from here -- can you write an example using only the public API that generates the above "bad" error message?

Certainly, I'll begin by locating an example that triggers this message. However, what if the error cannot be replicated using the public API? Will we simply let it remain unresolved?

**Comment by MichaelChirico** [comment]
Yes. We shouldn't waste time over-complicating an internal function that has edge case behavior no user can ever see (unless they "violate the contract" and access the private API).
